tests/Unit/Domain.LinnApps.Tests/ShortagesReportTests/WhenGettingReport.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.Tests.ShortagesReportTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+
+    using FluentAssertions;
+
+    using Linn.Common.Reporting.Models;
+    using Linn.Purchasing.Domain.LinnApps.Reports.Models;
+
+    using NSubstitute;
+
+    using NUnit.Framework;
+
+    public class WhenGettingReport : ContextBase
+    {
+        private IEnumerable<ResultsModel> results;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.results = this.Sut.GetReport(3, "ALL");
+        }
+
+
+        [Test]
+        public void ShouldGetData()
+        {
+            this.ShortagesRepository.Received().FilterBy(Arg.Any<Expression<Func<ShortagesEntry, bool>>>());
+        }
+
+        [Test]
+        public void ShouldReturnData()
+        {
+            this.results.Should().HaveCount(2);
```

Comment:
Since your report domain service is doing some grouping and totalling etc. I think you should test the correct results are actually built here (examples of checking report data here https://github.com/linn/purchasing/blob/main/tests/Unit/Domain.LinnApps.Tests/PurchaseOrdersReportServiceTests/WhenGettingOrdsBySupplierReportStockN.cs

You are also filtering your data in your domain service so you should have tests for different options scenarios e.g.
WhenPurchaseLevel3 or WhenVendorManagerL